resources/services/iam/users.go

```diff
@@ -657,7 +657,17 @@ func getCredentialReport(ctx context.Context, meta schema.ClientMeta) (reportUse
 		case "ReportNotPresent", "ReportExpired":
 			_, err := svc.GenerateCredentialReport(ctx, &iam.GenerateCredentialReportInput{})
 			if err != nil {
-				return nil, diag.WrapError(err)
+				var serviceError smithy.APIError
+				if !errors.As(err, &serviceError) {
+					return nil, diag.WrapError(err)
+				}
+				// LimitExceeded is the only specific error that should not stop processing
+				// If Limit Exceeded is returned we should try and see if there is a credential report
+				// already generated so we want to sleep for 5 seconds then continue
+				if serviceError.ErrorCode() != "LimitExceeded" {
+					return nil, diag.WrapError(err)
+				}
+				time.Sleep(5 * time.Second)
```

Comment:
Would add a `select` with `<-ctx.Done()` and a `<-time.After` to prevent 5sec hangs on interrupt.

Reply:
I updated to use the SDK helper you introduced